nwb_conversion_tools/datainterfaces/ecephys/neuralynx/neuralynxdatainterface.py

```diff
@@ -1,11 +1,80 @@
-"""Authors: Cody Baker."""
+"""Authors: Cody Baker and Ben Dichter."""
+import warnings
 from pathlib import Path
 from natsort import natsorted
+from glob import glob
+from dateutil import parser
+import json
 
 from spikeextractors import MultiRecordingChannelExtractor, NeuralynxRecordingExtractor
 
 from ..baserecordingextractorinterface import BaseRecordingExtractorInterface
 from ....utils import FolderPathType
+from ....utils.json_schema import dict_deep_update
+
+
+def parse_header(header):
+    header_dict = dict()
+    for line in header.split("\n")[1:]:
+        if line:
+            if line[0] == "-":
+                key, val = line[1:].split(" ", 1)
+                header_dict[key] = val
+    return header_dict
+
+
+def get_metadata(folder_path):
+    """
+    Parse the header of one of the .ncs files to get the session start time (without
+    timezone) and the session_id.
+
+    Parameters
+    ----------
+    folder_path: str
+
+    Returns
+    -------
+    dict
+
+    """
+    csc_files = glob(folder_path + "/*.ncs") or glob(folder_path + "/*.Ncs")
+    fpath = csc_files[0]
```

Comment:
probably because the files are not in a specific order. We could `sort` them to ensure that we are reading the same file.

Though the fact that these files have different start times might be concerning in itself..

Reply:
Yes, I think sorting by name get us a consistent pipeline. I will take care of this.

We should keep an eye on future files to see how different the extracted times are when we have more than one `.nsc` file.